### ISSUE TITLE
Make the Page Down and Page Up buttons work when viewing PDFs

### ISSUE THREAD

**Issue by lsolesen** [learningequality/kolibri#10718]
## Observed behavior

Page Down and Page Up does not work when viewing PDFs.

Many PDFs that we use are slideshows. Therefore it would be nice, if you could use the Page Down and Page Up to switch to the next slide (next page) in the PDF.
…

## Errors and logs

…

## Expected behavior

…

## User-facing consequences

…

## Steps to reproduce

…

## Context

…

**Comment by radinamatic** [comment]
Hey @lsolesen , thank you for flagging this!

Could you please always fill in the **`Context`** section when you report the issues, it's there for a reason as it makes easier to debug the problem 😉 

```
Tell us about your environment, including:
 * Kolibri version
 * Operating system
 * Browser
```

Testing this on Kolibri demo site (which is at version 0.15.12) in Firefox on Linux, I am able to use both the up & down arrow key, and the PgUp & PgDown key, but the trick is to **have the PDF itself in focus**. In other words, you cannot use arrow and page keys right after the PDF page loads, but if you use the `Tab` key to put the PDF in focus, you will be able to move through the pages with the keyboard. Granted we could improve this, I'm thinking of maybe fixing the skip-to-content feature that is currently broken, but it should be doable even now, just with a few more key presses. I tried to record the demo to illustrate:


https://github.com/learningequality/kolibri/assets/1457929/cbe5a79e-5286-472f-a53c-22b802c80b74

**Comment by lsolesen** [comment]
Doh. Sorry about that. I am in 0.15.11 on Windows in Chrome and Edge. I
cannot bring the PDF im focus, so I can use the keyboard for navigating.

Installed 0.15.12 and it is the same. Tab only switches between the + - and
Exit Screen. Clicking on the PDF with the mouse does not put it into focus
either.

tor. 18. maj 2023 19.49 skrev Radina Matic ***@***.***>:

**Comment by pcenov** [comment]
Hi @radinamatic, after additionally testing this in Windows, Ubuntu and Mac I can confirm the following:
- Fullscreen: The up and down buttons are working correctly in both Windows and Ubuntu in all browsers
- Non-fullscreen: The up and down buttons are working correctly only in Firefox in both Windows and Ubuntu and only if you use the Tab key to put the pdf in focus
- The up and down buttons are not functioning in both the Mac app and Safari.

**Comment by GarvitSinghal47** [comment]
hi @radinamatic , if the issue is still up then i would love to work on it , can you assign it to me.

**Comment by MisRob** [comment]
Hi @GarvitSinghal47, thank you for your interest. Looking at @radinamatic's comment

> Granted we could improve this, I'm thinking of maybe fixing the skip-to-content feature that is currently broken, but it should be doable even now, just with a few more key presses.

I think this issue needs some decisions at first before anyone can start working on it. We recently added some more good first issues and help wanted issues so before we have clearer expectations here, I'd recommend picking another one.

**Comment by pcenov** [comment]
Hi @radinamatic, after additionally testing this in Kolibri 0.17 RC 2 in Windows, Ubuntu and Mac I can confirm the following:

- Fullscreen: The up and down buttons are working correctly in Chrome, Firefox, Edge and Safari in all OSs.
- Non-fullscreen: The up and down buttons are also working correctly in Chrome, Firefox, Edge and Safari in all OSs but only if you use the Tab key to put the pdf in focus as mouse clicking the pdf doesn't do the trick
- Mac app: The up and down buttons are not functioning in the Mac app. In the Mac app the 'Enter fullscreen' button is not working at all.

**Comment by pcenov** [comment]
Hi @radinamatic, have checked this one again in Kolibri 0.18.1 in Windows, Ubuntu and Mac I can confirm that the following is still valid:

- Fullscreen: The up and down buttons are working correctly in Chrome, Firefox, Edge and Safari in all OSs.
- Non-fullscreen: The up and down buttons are also working correctly in Chrome, Firefox, Edge and Safari in all OSs but only if you use the Tab key to put the pdf in focus as mouse clicking the pdf doesn't do the trick
- Mac app: The up and down buttons are not functioning in the Mac app. In the Mac app the 'Enter fullscreen' button is not working at all.

**Comment by radinamatic** [comment]
I believe the real blocker here that we should investigate and fix is that the keyboard navigation to scroll up and down the page is not working in the Mac app, and we should do some more thinking to decide where the `Skip to main content` link should land before approaching potential changes there.

cc @MisRob to evaluate whether this would be open for contribution or better in our backlog to triage.

**Comment by MisRob** [comment]
@radinamatic Thanks for checking on this. All issues into backlog first please. If we locate some ares during triaging we want to open for contribution, it can be done during that process. Generally, we won't be labeling issues for community anymore that are not part of projects we planned for. cc @marcellamaki
